library/std/src/sys/pal/unix/fd.rs

```diff
@@ -269,6 +250,33 @@ impl FileDesc {
         }
     }
 
+    // We support old MacOS, iOS, watchOS, tvOS and visionOS versions that do not have `preadv`.
+    // There is no `syscall` possible in these platform for following apple versions:
```

Comment:
I'm not certain this comment (or the one one about `pwritev`) is accurate anymore. Went about looking into the apple SDKs and they all have:
```c
ssize_t preadv(int, const struct iovec *, int, off_t) __DARWIN_NOCANCEL(preadv) __API_AVAILABLE(macos(11.0), ios(14.0), watchos(7.0), tvos(14.0));
ssize_t pwritev(int, const struct iovec *, int, off_t) __DARWIN_NOCANCEL(pwritev) __API_AVAILABLE(macos(11.0), ios(14.0), watchos(7.0), tvos(14.0));
```
[Here's the iOS SDK header](https://github.com/xybp888/iOS-SDKs/blob/caaa91364346465d0605d85db70031daab71b5d5/iPhoneOS17.3.1.sdk/usr/include/sys/uio.h#L97-L108) but they all look the same on my local SDK's. I found `preadv`/`pwritev` in the VisionOS tbd ("text-based stub" libraries)

Reply:
Agree that it's the same in the headers for each SDK, can be checked using:
```bash
for sdk in macosx appletvos iphoneos watchos xros
do
    cat $(xcrun --show-sdk-path --sdk $sdk)/usr/include/sys/uio.h | grep API_AVAILABLE
done
```

That said, the comment is indeed correct that we still support e.g. macOS 10.12, while the API has only added in macOS 11. So we still need to do weak linking here.